### PR TITLE
[Docs] Add diagram about hedging's context and callbacks

### PR DESCRIPTION
**Pull request by peter-csala**
# Pull Request

## The issue or feature being addressed

- #1749

## Details on the issue fix or feature implementation

- As it was discussed in the review comments: it makes sense to visualize the sequence of actions related to context and callbacks
- @martintmk Can you please double check whether the diagram depicts everything and in the right order?
- [Preview](https://github.com/peter-csala/Polly/blob/add-sequence-diagram-about-hedgings-contexts-and-callbacks/docs/strategies/hedging.md#sequence-diagram-about-contexts-and-callbacks)

## Confirm the following

- [x]  I started this PR by branching from the head of the default branch
- [x]  I have targeted the PR to merge into the default branch
- [x]  I have included unit tests for the issue/feature
- [x]  I have successfully run a local build

### CODE REVIEW

File: docs/strategies/hedging.md [hunk 1]
```diff
@@ -471,6 +471,45 @@ Some insights about the hedging callback behavior in relation to hedging context
...
+    P->>H: Calls ExecuteCore <br/>with Primary Context
+    H-->>H: Deep clones <br/>Primary Context
+
+    H->>+UC: Invokes <br/>with Primary Context
+    UC-->>UC: Processes <br/>+ Modifies Context
+    UC->>-H: Fails
+
+    H-->>H: Deep clones <br/>Primary Context
+    H->>+AG: Invokes <br/>with both Contexts
+    AG-->>AG: Modifies Primary Context
```
Comment (martintmk) [inline docs/strategies/hedging.md:497]:
```suggestion
    AG-->>AG: Modifies Primary or Action Context 
```

you can also modify action context here.

File: docs/strategies/hedging.md [hunk 2]
```diff
@@ -471,6 +471,45 @@ Some insights about the hedging callback behavior in relation to hedging context
...
+    participant H as Hedging
+    participant AG as ActionGenerator
+    participant UC as UserCallback
+    participant HUC as HedgedUserCallback
+    participant OH as OnHedging
+
+    P->>H: Calls ExecuteCore <br/>with Primary Context
+    H-->>H: Deep clones <br/>Primary Context
+
+    H->>+UC: Invokes <br/>with Primary Context
```
Comment (martintmk) [inline docs/strategies/hedging.md:491]: note: in this call the Primary = Action context, they are filled but these are the same instances.

But I am thinking, is this OK? 
Shall we have the same behavior even for primary action? Just clone the primary context and use it for primary callback too. wdyt?

File: docs/strategies/hedging.md (hunk 1 above)
Comment (peter-csala) [reply docs/strategies/hedging.md]: I've updated the `OnHedging` part also.

File: docs/strategies/hedging.md (hunk 2 above)
Comment (peter-csala) [reply docs/strategies/hedging.md:491]: I think that approach would be cleaner as the original request won't be treated in a special way.

Comment (martintmk) [reply docs/strategies/hedging.md:491]: True, I have prepared the PR with the changes: #1754 

Can you also align the diagram with it?

**Comment by martintmk** [comment]
Some real-case scenario where this is used:

https://github.com/dotnet/extensions/blob/3038d565967654ffb92cdd5edf434930c9249e91/src/Libraries/Microsoft.Extensions.Http.Resilience/Hedging/ResilienceHttpClientBuilderExtensions.Hedging.cs#L84

File: docs/strategies/hedging.md (hunk 2 above)
Comment (peter-csala) [reply docs/strategies/hedging.md]: Done

**Review by martintmk — APPROVED** [review]

**Pull request merged**